scripts/fbt_tools/crosscc.py

```diff
@@ -15,7 +16,7 @@
 def prefix_commands(env, command_prefix, cmd_list):
     for command in cmd_list:
         if command in env:
-            env[command] = command_prefix + env[command]
+            env[command] = shutil.which(command_prefix + env[command])
```

Comment:
If that's for generating absolute paths in the .json file - it shouldn't be done like that. 
See `COMPILATIONDB_USE_ABSPATH` parameter for `compilation_db` tool.

Reply:
We have a fork of compilation_db tool, I think it would be better to implement that within it, without using absolute tool paths everywhere.

Anyway, regarding this PR: I couldn't make it work under Windows. clangd seems to be up and running, but it doesn't load the compilation database from /build/latest. What's your configuration where this works?

Also please add clangd extension to recommended plugins configuration shipped with fbt, and set cpptools as "unwanted" in it.

Reply:
> Anyway, regarding this PR: I couldn't make it work under Windows. clangd seems to be up and running, but it doesn't load the compilation database from /build/latest. What's your configuration where this works?

My setup is Linux+VSCodium+vscode-clangd and just loading the project under that.

Are there any indications in the log of what went wrong?

> Also please add clangd extension to recommended plugins configuration shipped with fbt, and set cpptools as "unwanted" in it.

Not sure if we want to move to clangd wholesale. Maybe we should have an alternative config that does what you say?

Reply:
OK so I dug deeper into this and turns out it's way, way more complicated than I initially thought.

`--query-driver` option expects a fnmatch pattern and it's extremely fragile in terms of handling path separators as everything needs to be prefect or it would refuse to match correctly (and that `${pathSeparator}` doesn't work for some reason). I don't see a good way of implementing this other than generating it from fbt, which is kind of intrusive and I don't really know much about scons to pull this off confidently.

As to the absolute tool path, looks like the `compilation_db` just takes pre-built commands from elsewhere and use them, which are a pain to work with, and doing a string replacement isn't a real option here. So looks like the best way would still be providing absolute tool paths to scons from the very beginning (or somehow make clangd resolve extra `PATH` set by fbt without writing absolute paths to `compile_commands` which is like, good luck.)

Reply:
> Anyway, regarding this PR: I couldn't make it work under Windows. clangd seems to be up and running, but it doesn't load the compilation database from /build/latest. What's your configuration where this works?
> 
> Also please add clangd extension to recommended plugins configuration shipped with fbt, and set cpptools as "unwanted" in it.

Addressed.

For point 1 I still think that changing the tool path from the very beginning seems to be the cleanest approach here.